static/panes/diff.ts

```diff
@@ -285,20 +273,35 @@ export class Diff extends MonacoPane<monaco.editor.IStandaloneDiffEditor, DiffSt
                 render: <any>{
                     option: (item, escapeHtml) => {
                         const origin = item.editorId !== false ? 'Editor #' + item.editorId : 'Tree #' + item.treeId;
+                        const editor = hub.getEditorById(item.editorId);
+                        const name = editor !== undefined ? editor.getPaneName() : origin;
                         return (
                             '<div>' +
-                            `<span class="compiler">${escapeHtml(item.compiler.name)}</span>` +
-                            `<span class="options">${escapeHtml(item.options)}</span>` +
+                            `<span class="paneName">${escapeHtml(name)}</span>` +
                             '<ul class="meta">' +
-                            `<li class="editor">${escapeHtml(origin)}</li>` +
-                            `<li class="compilerId">${escapeHtml(getItemDisplayTitle(item))}</li>` +
+                            `<li class="compiler">Compiler: ${escapeHtml(item.compiler.name)}</li>` +
+                            `<li class="options">Options: ${escapeHtml(item.options)}</li>` +
                             '</ul>' +
                             '</div>'
                         );
                     },
                 },
                 dropdownParent: 'body',
                 plugins: ['input_autogrow'],
+                onFocus: () => {
+                    instance.clear();
+                    instance.clearOptions();
+
+                    Object.entries(this.compilers).forEach(([_, entry]) => {
+                        const editor = this.hub.getEditorById(entry.id as number);
+                        if (editor !== undefined) {
+                            entry.name = editor.getPaneName();
+                        }
+                        instance.addOption(entry);
+                    });
```

Comment:
I do this so that if you rename a pane after creating the diff view you don't have to reload the page to see the changes take effect.